### DISCUSSION TITLE
Docker container keeps toggling between "starting" and "running"

### DISCUSSION THREAD

**Discussion by Ro4cHii** [OpenCCU/OpenCCU#1861] [🇺🇸 Q&A]
Hello everyone,

yesterday, I did a fresh installation of **Raspberry Pi OS Lite (32 bit)** on my **Raspberry Pi 1 Model B**.

I installed Docker, configured the API for external access via TCP, added it as environment in my portainer instance running on another system and successfully installed a first container.

Now, I tried to do the same for RasperryMatic following the steps mentioned in https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI.

Every step has been performed without any mentionable warning/error but still, the Docker container simply won't start correctly.

It appears as if it was starting and after some seconds, it changes its status to "running". 
Only to go back again to "starting" after a couple of seconds/minutes. The status seems to toggle in endless cycles. 
Regardless of the status, the web interface remains unavailable.

![image](https://user-images.githubusercontent.com/20772803/174435481-18b6d90f-7f46-45c0-b0ba-1e25498ffda3.png)
![image](https://user-images.githubusercontent.com/20772803/174435493-ec9a3217-ed6a-483f-ab0e-24a668565832.png)

The container status is "running for a few seconds [green heart]" and container health is "unhealthy [red heart]" with the **failure count** being **0**.

If I check my Docker logs by `docker logs homematic-ccu` nothing at all is output. The mapped volume folder "ccu_data" remains empty as well.

This is my setup:
* Raspberry Pi 1 Model B
* Raspberry Pi OS Lite (32 bit) [Linux raspberrypi 5.15.32+ #'1538 Thu Mar 31 19:37:58 BST 2022 armv61]
* HM-MOD-RPI-PCB

**docker-compose.yaml**
```
version: "3.8"
services:
  raspberrymatic:
    image: ghcr.io/jens-maus/raspberrymatic:latest
    container_name: homematic-ccu
    hostname: homematic-raspi
    privileged: true
    restart: unless-stopped
    stop_grace_period: 30s
    volumes:
      - /volume1/docker/homematic/ccu_data:/usr/local:rw
      - /lib/modules:/lib/modules:ro
      - /run/udev/control:/run/udev/control
    ports:
      - "8080:80"
      - "2001:2001"
      - "2010:2010"
      - "9292:9292"
      - "8181:8181"
volumes:
  ccu_data:
```

RaspberryMatic used to be working like a charm when I had installed the RaspberryMatic OS but to be more flexible with the host system, I wanted to switch to using a Docker container.

What am I doing wrong?

**Comment by oradke** [comment]
You're posting in the wrong forum. Check here first:
https://homematic-forum.de/forum/

**Reply by jens-maus** [reply]
> You're posting in the wrong forum. Check here first: https://homematic-forum.de/forum/

Who says that this is the wrong forum? It is perfectly fine to use this GitHub discussion forum as well!

**Reply by oradke** [reply]
OH, OK. I just figured this is just for bugs and development discussions. My bad, I apologize.

**Reply by jens-maus** [reply]
> OH, OK. I just figured this is just for bugs and development discussions. My bad, I apologize.

You are mixing up the [Discussions](https://github.com/jens-maus/RaspberryMatic/discussions) area here in GitHub with the [Issues](https://github.com/jens-maus/RaspberryMatic/issues) area which of course is only for reporting bugs and feature enhancement requests, but the "Discussions" section is of course perfectly fine to be used for discussions ;)

**Reply by oradke** [reply]
AHHH! OK, got it. Good to know. Thanks!

**Comment by Ro4cHii** [comment]
One addition:

In my portainer's event log I can see that the homematic-ccu container exited with status code 132.
I checked some discussions I have found on Google and people are indicating that it may be a problem of the CPU lacking AVX support or that the container has been built on another system (architecture incompatibility).

In the above-mentioned guide it says that "It is possible to run RaspberryMatic also as a docker/OCI container on existing amd64, aarch64/arm64 or arm-based systems." May this be the problem that I am using a 32bit OS?

**Reply by jens-maus** [reply]
Didn't test this rare setup here yet. But from my feelings I could already say that using these old 32bit 512MB Pi1s for something like running docker containers is overpushing it over its limits! Thus, you should consider using at least a Pi3 for that purpose!

**Reply by Ro4cHii** [reply]
If I am lucky I have an Odroid C2 here which is 64bit and a little bit more powerful. Can you say if the HM-MOD-RPI-PCB will work on these too (without rewiring GPIO etc.)?

**Reply by jens-maus** [reply]
> If I am lucky I have an Odroid C2 here which is 64bit and a little bit more powerful. Can you say if the HM-MOD-RPI-PCB will work on these too (without rewiring GPIO etc.)?

Yes, should work.

**Reply by Ro4cHii** [reply]
I have installed DietPi on my Odroid C2 now but unfortunately, I am failing on step 7 of the [guide](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI).

Any idea what I can do?

```
root@DietPi:/etc/modules-load.d# sudo modprobe eq3_char_loop
modprobe: FATAL: Module eq3_char_loop not found in directory /lib/modules/5.10.110-meson64
```

Obviously I cannot do what the guide suggests (`sudo apt install raspberrypi-kernel raspberrypi-kernel-headers raspberrypi-bootloader`).

**Reply by jens-maus** [reply]
Well, as you are not using a raspberrypi you obviously have to take care that the DietPi relatved kernel+kernel headers are installed via apt. Thus, search in google on how to install these kernel+kernel headers in DietPi so that the `pivccu-modules-dkms` package (if re-installed) can compile all necessary kernel modules.

**Reply by jens-maus** [reply]
... thus, try to use the following apt command to install the kernel headers:

```
sudo apt install linux-headers-$(uname -r)
```

and then reinstall the pivccu-modules-dkms package.

**Reply by jens-maus** [reply]
So did my suggestion work for you now, or not?

**Reply by Ro4cHii** [reply]
No, unfortunately it did not.

When I try to execute that command, I receive the error message
```
E: Unable to locate package linux-headers-3.16.85-65
E: Couldn't find any package by glob 'linux-headers-3.16.85-65'
```

I tried to install a newer kernel following three different tutorials but either I did something wrong or these Odroid builds are not able to update the kernel - no matter what I do, the kernel remains the same.

I also tried to use some of the other package names suggested in https://forum.odroid.com/viewtopic.php?t=24590 but the result's the same. [Here](https://forum.odroid.com/viewtopic.php?t=33648) I read that the kernel headers are actually supposed to be located in `/lib/modules/$(uname -r)/build/include/` and indeed there are some header files there. Someone suggested to create a symbolic link in `/usr/src/` that points there but nonetheless, I always end up receiving the info that Module eq3_char_loop could not be found in `/lib/modules/3.16.85-65` (before it was looking in `/lib/modules/5.10.110-meson64`).

**Reply by Ro4cHii** [reply]
I just installed yet another distribution; Armbian Bullseye.

Starting from here, a much newer kernel is already installed (5.10.110-meson64).
But again, I get the same error message when I try to execute `sudo modprobe eq3_char_loop`.

I now installed `linux-headers-current-meson64 22.05.03` (during installation, it said that it was compiling headers etc.) and the re-executed the commands 

`sudo apt install pivccu-modules-dkms` --> `pivccu-modules-dkms is already the newest version (1.0.77).` 
and
`sudo modprobe eq3_char_loop` --> `modprobe: FATAL: Module eq3_char_loop not found in directory /lib/modules/5.10.110-meson64`

I also installed `linux-headers-edge-meson64 22.05.01` (these two seemed to be the only two available ones to fit as far as possible but eventually they do not fit entirely...) - same result.

In `/usr/src` there are four folders now but obviously there seems to be a version mismatch between kernel and headers:

- `5.10.110-meson64`
- `linux-headers-5.17.5-meson64`
- `linux-headers-5.10.123-meson64`
- `pivccu-1.0.77`

If now I use auto completion on `sudo apt install linux-headers-5.10.` it automatically completes to `linux-headers-5.10.0-1` which already seems to be deviating from my kernel version.

So basically my understanding is the headers don't fit the kernel and to be honest, at this point I don't even know anymore what exactly I am actually doing ...

**Reply by Ro4cHii** [reply]
Ok I think I made some progress but I'm not quite there yet.

I did a fresh install of Armbian Bullseye and used `armbian-config` to install the kernel headers - they should match the used kernel now.
In `/usr/src` there are two folders now:
* `linux-headers-5.10.123-meson64`
* `pivccu-1.0.77`

In `/lib/modules` there's one:
* `5.10.123-meson64` 

I rebooted after installing the kernel headers and executed all the steps from the quick guide but still, I receive the same error message:
`modprobe: FATAL: Module eq3_char_loop not found in directory /lib/modules/5.10.123-meson64`.

The output from the `sudo apt install pivccu-modules-dkms` command shows that there is an error in make.log:

```
root@odroidc2:~# sudo apt install pivccu-modules-dkms
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following NEW packages will be installed:
  pivccu-modules-dkms
0 upgraded, 1 newly installed, 0 to remove and 5 not upgraded.
Need to get 42.1 kB of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 https://www.pivccu.de/piVCCU stable/main arm64 pivccu-modules-dkms all 1.0.77 [42.1 kB]
Fetched 42.1 kB in 0s (110 kB/s)
Preconfiguring packages ...
Selecting previously unselected package pivccu-modules-dkms.
(Reading database ... 68722 files and directories currently installed.)
Preparing to unpack .../pivccu-modules-dkms_1.0.77_all.deb ...
Unpacking pivccu-modules-dkms (1.0.77) ...
Setting up pivccu-modules-dkms (1.0.77) ...
Error! The module/version combo: pivccu-1.0.77
is not located in the DKMS tree.
  SYNC    include/config/auto.conf.cmd
  CC      scripts/mod/empty.o
  HOSTCC  scripts/mod/mk_elfconfig
  MKELF   scripts/mod/elfconfig.h
  HOSTCC  scripts/mod/modpost.o
  CC      scripts/mod/devicetable-offsets.s
  HOSTCC  scripts/mod/file2alias.o
  HOSTCC  scripts/mod/sumversion.o
  HOSTLD  scripts/mod/modpost
scripts/Makefile.build:415: warning: overriding recipe for target 'modules.order'
Makefile:1422: warning: ignoring old recipe for target 'modules.order'
cc     recordmcount.c   -o recordmcount

Creating symlink /var/lib/dkms/pivccu/1.0.77/source ->
                 /usr/src/pivccu-1.0.77

DKMS: add completed.

Kernel preparation unnecessary for this kernel.  Skipping...

Building module:
cleaning build area...
make -j4 KERNELRELEASE=5.10.123-meson64 ARCH=arm64 all........(bad exit status: 2)

=====================================================
Error! Bad return status for module build on kernel: 5.10.123-meson64 (aarch64)
Consult /var/lib/dkms/pivccu/1.0.77/build/make.log for more information.
=====================================================

Created symlink /etc/systemd/system/multi-user.target.wants/pivccu-dkms.service → /lib/systemd/system/pivccu-dkms.service.
Created symlink /etc/systemd/system/debmatic.service.wants/pivccu-dkms.service → /lib/systemd/system/pivccu-dkms.service.
Created symlink /etc/systemd/system/pivccu.service.requires/pivccu-dkms.service → /lib/systemd/system/pivccu-dkms.service.
root@odroidc2:~# sudo sh -c 'echo eq3_char_loop >/etc/modules-load.d/eq3_char_loop.conf'
root@odroidc2:~# sudo service pivccu-dkms start
root@odroidc2:~# sudo modprobe eq3_char_loop
modprobe: FATAL: Module eq3_char_loop not found in directory /lib/modules/5.10.123-meson64

```

Contents of make.log:
```

DKMS make.log for pivccu-1.0.77 for kernel 5.10.123-meson64 (aarch64)
Thu 30 Jun 2022 11:24:46 PM CEST
make -C /lib/modules/5.10.123-meson64/build M=/var/lib/dkms/pivccu/1.0.77/build modules
make[1]: Entering directory '/usr/src/linux-headers-5.10.123-meson64'
  CC [M]  /var/lib/dkms/pivccu/1.0.77/build/eq3_char_loop.o
  CC [M]  /var/lib/dkms/pivccu/1.0.77/build/plat_eq3ccu2.o
  CC [M]  /var/lib/dkms/pivccu/1.0.77/build/generic_raw_uart.o
  CC [M]  /var/lib/dkms/pivccu/1.0.77/build/pl011_raw_uart.o
  CC [M]  /var/lib/dkms/pivccu/1.0.77/build/dw_apb_raw_uart.o
  CC [M]  /var/lib/dkms/pivccu/1.0.77/build/meson_raw_uart.o
  CC [M]  /var/lib/dkms/pivccu/1.0.77/build/fake_hmrf.o
  CC [M]  /var/lib/dkms/pivccu/1.0.77/build/rpi_rf_mod_led.o
  CC [M]  /var/lib/dkms/pivccu/1.0.77/build/dummy_rx8130.o
  CC [M]  /var/lib/dkms/pivccu/1.0.77/build/led_trigger_timer.o
  CC [M]  /var/lib/dkms/pivccu/1.0.77/build/hb_rf_usb.o
/var/lib/dkms/pivccu/1.0.77/build/fake_hmrf.c: In function ‘fake_hmrf_write’:
/var/lib/dkms/pivccu/1.0.77/build/fake_hmrf.c:268:1: warning: the frame size of 2080 bytes is larger than 1024 bytes [-Wframe-larger-than=]
  268 | }
      | ^
  CC [M]  /var/lib/dkms/pivccu/1.0.77/build/hb_rf_usb_2.o
  CC [M]  /var/lib/dkms/pivccu/1.0.77/build/hb_rf_eth.o
  CC [M]  /var/lib/dkms/pivccu/1.0.77/build/rtc-rx8130.o
/var/lib/dkms/pivccu/1.0.77/build/hb_rf_eth.c: In function ‘hb_rf_eth_init’:
/var/lib/dkms/pivccu/1.0.77/build/hb_rf_eth.c:775:3: warning: ignoring return value of ‘sysfs_create_file’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  775 |   sysfs_create_file(&dev->kobj, &dev_attr_is_connected.attr);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/lib/dkms/pivccu/1.0.77/build/hb_rf_eth.c:776:3: warning: ignoring return value of ‘sysfs_create_file’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  776 |   sysfs_create_file(&dev->kobj, &dev_attr_connect.attr);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  MODPOST /var/lib/dkms/pivccu/1.0.77/build/Module.symvers
make[3]: *** No rule to make target 'scripts/module.lds', needed by '/var/lib/dkms/pivccu/1.0.77/build/dummy_rx8130.ko'.  Stop.
make[3]: *** Waiting for unfinished jobs....
  CC [M]  /var/lib/dkms/pivccu/1.0.77/build/dummy_rx8130.mod.o
make[2]: *** [scripts/Makefile.modpost:130: __modpost] Error 2
make[1]: *** [Makefile:1726: modules] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-5.10.123-meson64'
make: *** [Makefile:30: all] Error 2
```

Any idea what went wrong here?

**Reply by Ro4cHii** [reply]
Hooray - I finally managed to get it to run on my Odroid C2!

I certainly don't know the solution to the problem mentioned in my last comment but I figured that it might work with an older kernel so I switched from `linux-5.10.123-meson64` to `linux-5.10.60-meson64`.

With Armbian, it doesn't seem to be possible to install other kernel headers than 5.10.110 and 5.10.123 via `armbian-config`.
So as suggested somewhere in the Armbian forum, I had to download them from `https://imola.armbian.com/apt/pool/main/l/linux-5.10.60-meson64/` and install them myself.

With the older kernel, I was able to execute the modprobe command and no error message was thrown.

The docker container is running, the web interface is reachable and I expect that in terms of functionality everything should be working as well.

If I will not update this thread in the next 2-3 days, you can expect that I reached my goal 😃

Thank you very much for your help so far!

**Reply by Ro4cHii** [reply]
Well - this would've been too good to be true.

When I click on "teach-in devices" in the WebUI, it says:
`No HomeMatic radio module or LAN gateway was found to be usable for homematic or homematicIP communication in your system. Please make sure that a compatible rf module or LAN gateway is correctly connected or configured so that you are able to teach-in devices.`.

The commands `sudo apt install pivccu-modules-dkms`, `sudo service pivccu-dkms start` and `sudo modprobe eq3_char_loop` were executed without any mentionable log message.

This is my docker-compose.yaml
```
version: "3.8"
services:
  raspberrymatic:
    image: ghcr.io/jens-maus/raspberrymatic:latest
    container_name: ccu
    hostname: homematic-raspi
    privileged: true
    restart: unless-stopped
    stop_grace_period: 30s
    volumes:
      - /root/ccu_data:/usr/local:rw
      - /lib/modules:/lib/modules:ro
      - /run/udev/control:/run/udev/control
    ports:
      - "8080:80"
      - "2001:2001"
      - "2010:2010"
      - "9292:9292"
      - "8181:8181"
volumes:
  ccu_data:
```

And here is the RaspberryMatic log file:
```
***** messages *****
Jul  4 00:23:56 homematic-raspi syslog.info syslogd started: BusyBox v1.35.0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ vgic(9) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ arch_timer(11) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ vtimer(12) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ arm-pmu(14) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ arm-pmu(15) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ arm-pmu(16) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ arm-pmu(17) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ c1108500.i2c(20) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ c1108680.adc(21) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ ttyAML0(23) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ c8100580.ir(24) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ vdec(25) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ esparserirq(26) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ platform_mhu_link(28) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ platform_mhu_link(29) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ eth0(30) guessed as class 5
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ d0072000.mmc(31) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ d0074000.mmc(32) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ gp(33) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ gpmmu(34) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ pp_bcast(35) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ pp0(37) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ ppmmu0(38) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ pp1(39) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ ppmmu1(40) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ pp2(41) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ ppmmu2(42) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ meson(43) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ c883a000.hdmi-tx(44) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ dwc2_hsotg:usb1(45) guessed as class 0
Jul  4 00:23:56 homematic-raspi daemon.debug /usr/sbin/irqbalance: IRQ stmmac-0:00(46) guessed as class 0
Jul  4 00:23:57 homematic-raspi user.info firewall: configuration set
Jul  4 00:23:59 homematic-raspi daemon.err xinetd[310]: Unable to read included directory: /etc/config/xinetd.d [file=/etc/xinetd.conf] [line=14]
Jul  4 00:23:59 homematic-raspi daemon.crit xinetd[310]: 310 {init_services} no services. Exiting...
Jul  4 00:24:08 homematic-raspi local0.err ReGaHss: ERROR: original file not exists, so try load new file. [LoadOM():iseDOM.cpp:2610]
Jul  4 00:24:08 homematic-raspi local0.err ReGaHss: ERROR: new file not exists, so try load bak file. [LoadOM():iseDOM.cpp:2622]
Jul  4 00:24:08 homematic-raspi local0.err ReGaHss: ERROR: failed open file= /etc/config/homematic.regadom.bak [LoadFromFile():iseDOMpersist.cpp:78]
Jul  4 00:24:08 homematic-raspi local0.err ReGaHss: ERROR: (202) failed! Already in map! [Insert():iseDOMobj.cpp:517]
Jul  4 00:24:08 homematic-raspi local0.err ReGaHss: ERROR: IseAddFavorite failed! ID=202 [IseAddFavorite():iseDOM.cpp:644]
Jul  4 00:24:08 homematic-raspi local0.err ReGaHss: ERROR: (203) failed! Already in map! [Insert():iseDOMobj.cpp:517]
Jul  4 00:24:08 homematic-raspi local0.err ReGaHss: ERROR: IseAddFavorite failed! ID=203 [IseAddFavorite():iseDOM.cpp:644]
Jul  4 00:24:08 homematic-raspi local0.err ReGaHss: ERROR: (204) failed! Already in map! [Insert():iseDOMobj.cpp:517]
Jul  4 00:24:08 homematic-raspi local0.err ReGaHss: ERROR: IseAddFavorite failed! ID=204 [IseAddFavorite():iseDOM.cpp:644]
Jul  4 00:24:10 homematic-raspi user.info homematic: meine-homematic.de Versions - Upgrade auf 1.503
Jul  4 00:24:10 homematic-raspi user.info homematic: meine-homematic.de VPN Dienst und Autoupdate DEAKTIVIERT
Jul  3 22:24:12 homematic-raspi daemon.info : starting pid 550, tty '': '/bin/mv /tmp/boot.log /var/log/boot.log'
Jul  3 22:24:12 homematic-raspi daemon.info : starting pid 551, tty '/dev/null': '/usr/bin/monit -Ic /etc/monitrc'
Jul  4 00:24:12 homematic-raspi user.info monit[551]:  New Monit id: b10eb028d5f8d4bb384e9992406e130f  Stored in '/usr/local/etc/monit.id'
Jul  4 00:24:12 homematic-raspi user.info monit[551]: Starting Monit 5.30.0 daemon with http interface at /var/run/monit.sock
Jul  4 00:24:12 homematic-raspi user.info monit[551]: 'homematic-raspi' Monit 5.30.0 started
Jul  4 00:24:25 homematic-raspi user.err monit[551]: 'sshdEnabled' status failed (1) -- no output
Jul  4 00:24:25 homematic-raspi user.err monit[551]: 'hs485dEnabled' status failed (2) -- grep: /var/etc/hs485d.conf: No such file or directory
Jul  4 00:24:25 homematic-raspi user.err monit[551]: 'multimacdEnabled' status failed (1) -- no output
Jul  4 00:24:25 homematic-raspi user.err monit[551]: 'hmlangwEnabled' status failed (1) -- no output
Jul  4 00:24:25 homematic-raspi user.err monit[551]: 'rfdEnabled' status failed (1) -- no output
Jul  4 00:24:25 homematic-raspi user.warn monit[551]: 'hasUSB' status failed (1) -- no output
Jul  4 00:24:25 homematic-raspi user.err monit[551]: 'hb_rf_eth-CheckEnabled' status failed (1) -- no output
Jul  4 00:24:25 homematic-raspi user.err monit[551]: 'coProcessorCheck' status failed (1) -- no output
Jul  4 00:24:25 homematic-raspi user.warn monit[551]: 'rpi4usb3Check' status failed (1) -- no output
Jul  4 00:24:25 homematic-raspi user.err monit[551]: 'tailscaleEnabled' status failed (1) -- no output
Jul  4 00:24:25 homematic-raspi user.err monit[551]: Lookup for '/media/usb1' filesystem failed  -- not found in /proc/self/mounts
Jul  4 00:24:25 homematic-raspi user.err monit[551]: Filesystem '/media/usb1' not mounted
Jul  4 00:24:25 homematic-raspi user.err monit[551]: 'usb1' unable to read filesystem '/media/usb1' state
Jul  4 00:24:25 homematic-raspi user.info monit[551]: 'usb1' trying to restart
Jul  4 00:24:41 homematic-raspi user.warn monit[551]: 'hasUSB' status failed (1) -- no output
Jul  4 00:24:41 homematic-raspi user.warn monit[551]: 'rpi4usb3Check' status failed (1) -- no output
Jul  4 00:24:41 homematic-raspi user.err monit[551]: Filesystem '/media/usb1' not mounted
Jul  4 00:24:41 homematic-raspi user.err monit[551]: 'usb1' unable to read filesystem '/media/usb1' state
Jul  4 00:24:41 homematic-raspi user.info monit[551]: 'usb1' trying to restart
Jul  4 00:24:57 homematic-raspi user.warn monit[551]: 'hasUSB' status failed (1) -- no output
Jul  4 00:24:57 homematic-raspi user.err monit[551]: 'rpi4usb3Check' status failed (1) -- no output
Jul  4 00:24:57 homematic-raspi user.err monit[551]: Filesystem '/media/usb1' not mounted
Jul  4 00:24:57 homematic-raspi user.err monit[551]: 'usb1' unable to read filesystem '/media/usb1' state
Jul  4 00:24:57 homematic-raspi user.info monit[551]: 'usb1' trying to restart
Jul  4 00:25:13 homematic-raspi user.warn monit[551]: 'hasUSB' status failed (1) -- no output
Jul  4 00:25:13 homematic-raspi user.err monit[551]: Filesystem '/media/usb1' not mounted
Jul  4 00:25:13 homematic-raspi user.err monit[551]: 'usb1' unable to read filesystem '/media/usb1' state
Jul  4 00:25:13 homematic-raspi user.info monit[551]: 'usb1' trying to restart
Jul  4 00:25:29 homematic-raspi user.err monit[551]: 'hasUSB' status failed (1) -- no output


***** hmserver.log *****
Jul 4 18:38:13 de.eq3.ccu.system.http.service.EnergyPriceController ERROR [vert.x-worker-thread-11] Can´t read the config file for the energy counter: /etc/config/energyPrice (No such file or directory) 
```

@jens-maus: Do you have an idea what could be the cause for this? Do I need to map something more (hardware device etc.)? I checked and compared the pinning of the Raspberry Pi B and the Odroid C2 - the module is installed correctly.

**Reply by Ro4cHii** [reply]
It's been a while.

Does maybe anyone else have an idea why the HM-MOD-RPI-PCB is not getting recognizd?

**Comment by Ro4cHii** [comment]
With a fresh setup, the Docker-related issues are solved.
It is still not working for me but my exact issue is described in https://github.com/jens-maus/RaspberryMatic/issues/1707.

This issue can be closed.
